analyzer/codechecker_analyzer/cmd/fixit.py

```diff
@@ -0,0 +1,349 @@
+        match_reports(diag['DiagnosticMessage']) and
+        check_existence_and_collect(diag['DiagnosticMessage']['FilePath']) and
+        check_modification_and_collect(diag['DiagnosticMessage']['FilePath']),
+        content['Diagnostics'])
+
+    if interactive:
+        items = filter(ask_user, items)
+
+    content['Diagnostics'] = list(items)
+
+    for diag in content['Diagnostics']:
+        existing_files.add(diag['DiagnosticMessage']['FilePath'])
+
+    return existing_files, not_existing_files, modified_files
+
+
+def list_fixits(inputs, checker_names, file_paths, interactive, reports):
+    """
+    This function dumps the .yaml files to the standard output like a "dry run"
+    with the replacements. See clang_tidy_fixit_filter() for the documentation
+    of the filter options.
+    inputs -- A list of report directories which contains the fixit dumps in a
+              subdirectory named "fixit".
+    """
+    not_existing_files = set()
+    existing_files = set()
+    modified_files = set()
+
+    for i in inputs:
+        fixit_dir = os.path.join(i, 'fixit')
+
+        if not os.path.isdir(fixit_dir):
+            LOG.info('No fixits in %s', i)
+            continue
+
+        for fixit_file in os.listdir(fixit_dir):
+            fixit_file = os.path.join(fixit_dir, fixit_file)
+            with open(fixit_file, encoding='utf-8', errors='ignore') as f:
+                content = yaml.load(f, Loader=yaml.BaseLoader)
+                fixit_mtime = util.get_last_mod_time(fixit_file)
+
+                existing, not_existing, modified = clang_tidy_fixit_filter(
+                    content, checker_names, file_paths, reports, fixit_mtime,
+                    interactive)
+
+                existing_files.update(existing)
+                not_existing_files.update(not_existing)
+                modified_files.update(modified)
+            print(yaml.dump(content))
+
+    if existing_files:
+        print("Updated files:\n{}".format(
+            '\n'.join(sorted(existing_files))),
+            file=sys.stderr)
+    if not_existing_files:
+        print("Not existing files:\n{}".format(
+            '\n'.join(sorted(not_existing_files))),
+            file=sys.stderr)
+    if modified_files:
+        print("Skipped files due to modification since last analysis:\n{}"
+              .format('\n'.join(sorted(modified_files))),
+              file=sys.stderr)
+
+
+def apply_fixits(inputs, checker_names, file_paths, interactive, reports):
+    """
+    This function applies the replacements from the .yaml files.
+    inputs -- A list of report directories which contains the fixit dumps in a
+              subdirectory named "fixit".
+    """
+    not_existing_files = set()
+    existing_files = set()
+    modified_files = set()
+
+    for i in inputs:
+        fixit_dir = os.path.join(i, 'fixit')
+
+        if not os.path.isdir(fixit_dir):
+            LOG.info('No fixits in %s', i)
```

Comment:
I analyzed tmux with the following command:
```sh
CodeChecker analyze ./build.json -o ./reports --analyzers clang-tidy -c
```
So I don't use the --enable-all option. In this case the fixit directory was created but it was empty. If I tried to list fixits in the report directory created above (`CodeChecker fixit -l ./reports/`) but the output was empty.